src/MeshlessFV/MfvCommon.cpp

```diff
@@ -45,19 +45,39 @@ using namespace std;
 /// MfvCommon class constructor.  Calls main SPH class constructor and also
 /// sets additional kernel-related quantities
 //=================================================================================================
-template <int ndim, template<int> class kernelclass>
-MfvCommon<ndim, kernelclass>::MfvCommon
+template <int ndim, template<int> class kernelclass, class SlopeLimiter>
+MfvCommon<ndim, kernelclass,SlopeLimiter>::MfvCommon
   (int hydro_forces_aux, int self_gravity_aux, FLOAT _accel_mult, FLOAT _courant_mult,
    FLOAT _h_fac, FLOAT h_converge_aux, FLOAT gamma_aux, string gas_eos_aux, string KernelName,
    int size_part, SimUnits &units, Parameters *params):
   MeshlessFV<ndim>(hydro_forces_aux, self_gravity_aux, _accel_mult, _courant_mult, _h_fac,
                    h_converge_aux, gamma_aux, gas_eos_aux, KernelName, size_part, units, params),
-  kern(kernelclass<ndim>(KernelName))
+  kern(kernelclass<ndim>(KernelName)),
+  riemannExact(gamma_aux, params->intparams["zero_mass_flux"]),
+  riemannHLLC(gamma_aux, params->intparams["zero_mass_flux"], gas_eos_aux == "isothermal")
 {
   this->kernp      = &kern;
   this->kernfac    = (FLOAT) 1.0;
   this->kernfacsqd = (FLOAT) 1.0;
   this->kernrange  = this->kernp->kernrange;
+
+  // Local references to parameter variables for brevity
+  map<string, string> &stringparams = params->stringparams;
+
+
+  // Riemann solver object
+  //-----------------------------------------------------------------------------------------------
+  string riemann_solver = stringparams["riemann_solver"];
+  if (riemann_solver == "exact") {
+    RiemannSolverType = 1 ;
```

Comment:
Yes, this would probably make this more readable if we left it as it is.  However, I was hoping at some point in the near future we could have a more adaptive Riemann solver, i.e. one which uses the linear/HLLC in most cases but only uses the exact solver in some cases where the approximate solvers break down.  So, if we do that soon, then there's no need to tidy this up too much.